plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -188,7 +188,7 @@ private static int allocatedProcessors(Settings settings) {
     }
 
     private static int urgentPoolCount(Settings settings) {
-        return boundedBy((allocatedProcessors(settings) + 7) / 8, 1, 2);
+        return boundedBy((allocatedProcessors(settings) + 1) / 2, 1, 2);
```

Comment:
sorry what does this change try to achieve for my understanding and is it used only on the context of remote publication

Reply:
With the previous formula (`boundedBy((allocatedProcessors(settings) + 7) / 8, 1, 2)`), the number of threads would be 2 when number of cores is 16 or more. For cores less than 1, number of threads would be 1.
With the new formula, the number of threads would be 2 when there are 4 cores or more.